src/revparse.c

```diff
@@ -899,9 +899,17 @@ int git_revparse(
 			rstr++;
 		}
 
-		error = git_revparse_single(&revspec->from, repo, lstr);
-		if (!error)
-			error = git_revparse_single(&revspec->to, repo, rstr);
+		error = git_revparse_single(
+			&revspec->from,
+			repo,
+			*lstr == '\0' ? "HEAD" : lstr);
+
+		if (!error) {
+			error = git_revparse_single(
+				&revspec->to,
+				repo,
+				*rstr == '\0' ? "HEAD" : rstr);
+		}
```

Comment:
Note that this also supports the ranges ".." and "...", that is no limiter at all. While these seem nonsensical, it even seems to be the right thing as git.git also accepts these ranges:

```
$ git rev-parse ..
..
$ git rev-parse ...
a8c90ef62281db933118aa84489eb0e1e9cc347c
a8c90ef62281db933118aa84489eb0e1e9cc347c
^a8c90ef62281db933118aa84489eb0e1e9cc347c
```

While the first result stumps me, there at least is no error here.

Could you maybe add another two tests for these special-cases?